main/models.py

```diff
@@ -195,17 +195,32 @@ def weeks_to_deadline(self) -> tuple[int, float] | None:
 
         return None
 
+    @property
+    def total_effort(self) -> int | None:
+        """Provide the total days worth of effort available from funding.
+
+        Returns:
+            The total number of days effort, or None if there is no funding information.
+        """
+        if self.funding_source.exists():
+            total = sum([funding.effort for funding in self.funding_source.all()])
+            return total
+
+        return None
+
     @property
     def days_left(self) -> tuple[int, float] | None:
         """Provide the days worth of effort left.
 
-        TODO: Placeholder. To be implemented when funding is implemented.
-
         Returns:
-            The number of days worth of effort left, or None if there is no funding
-            information.
+            The number of days and percentage worth of effort left, or None if there is
+            no funding information.
         """
-        return 42, 48.0
+        if self.funding_source.exists() and self.total_effort:
```

Comment:
If total_effort is not None, then funding_sources exist, so you don't need to check that as well.

Reply:
Thanks will add this change!